deepchecks/nlp/utils/text_properties.py

```diff
@@ -870,40 +917,41 @@ def calculate_builtin_properties(
-    )
-    for text in progress_bar:
-        progress_bar.set_postfix(
-            {'Sample': truncate_string(text, max_length=20) if text else 'EMPTY STRING'},
-            refresh=False
-        )
-        if pd.isna(text):
-            for prop in text_properties:
-                calculated_properties[prop['name']].append(np.nan)
-            continue
-        sample_language = run_available_kwargs(language, text=text, **kwargs)
+    for i in tqdm(range(0, len(raw_text), batch_size)):
+        batch = raw_text[i:i + batch_size]
+        batch_properties = defaultdict(list)
+
+        # filtering out empty sequences
+        filtered_sequences = [seq for seq in batch if pd.isna(seq) is False]
+
+        samples_language = _batch_wrapper(text_batch=filtered_sequences, func=language, **kwargs)
         if is_language_property_requested:
-            calculated_properties['Language'].append(sample_language)
-        kwargs['language_property_result'] = sample_language  # Pass the language property result to other properties
+            batch_properties['Language'].extend(samples_language)
+        kwargs['language_property_result'] = samples_language  # Pass the language property to other properties
 
         for prop in text_properties:
             if prop['name'] in import_warnings:  # Skip properties that failed to import:
-                calculated_properties[prop['name']].append(np.nan)
-            elif sample_language != 'en' and prop['name'] in english_properties_names \
-                    and ignore_non_english_samples_for_english_properties is True:
-                calculated_properties[prop['name']].append(np.nan)
+                batch_properties[prop['name']].extend([np.nan] * len(batch))
             else:
+                if prop['name'] in english_properties_names \
+                        and ignore_non_english_samples_for_english_properties is True:
+                    filtered_sequences = \
+                        [seq for seq, lang in zip(filtered_sequences, samples_language) if lang == 'en']
+                kwargs['batch_size'] = batch_size
                 try:
-                    value = run_available_kwargs(prop['method'], text=text, **kwargs)
-                    calculated_properties[prop['name']].append(value)
+                    if prop['name'] in BATCH_PROPERTIES:
+                        value = run_available_kwargs(func=prop['method'], text_batch=filtered_sequences, **kwargs)
+                    else:
+                        value = _batch_wrapper(text_batch=filtered_sequences, func=prop['method'], **kwargs)
+                    batch_properties[prop['name']].extend(value)
                 except ImportError as e:
                     warnings.warn(warning_message.format(prop['name'], str(e)))
-                    calculated_properties[prop['name']].append(np.nan)
+                    batch_properties[prop['name']].extend([np.nan] * len(batch))
                     import_warnings.add(prop['name'])
```

Comment:
Not sure I followed here. I don't see how you keep the order here. what if I have 32 samples, and sample number 7 is not in English? Where do you make sure that number 7 gets the "none" value? I only see that you filter by language but don't fill the others with None, and make sure to keep the order.

Reply:
Fixed.